kong/tracing/propagation.lua

```diff
@@ -447,11 +447,10 @@ local function set(conf_header_type, found_header_type, proxy_span, conf_default
   end
 
   if conf_header_type == "b3-single" or found_header_type == "b3-single" then
-    set_header("b3", fmt("%s-%s-%s-%s",
-        to_hex(proxy_span.trace_id),
-        to_hex(proxy_span.span_id),
-        proxy_span.should_sample and "1" or "0",
-      to_hex(proxy_span.parent_id)))
+    set_header("b3", to_hex(proxy_span.trace_id) ..
+        "-" .. to_hex(proxy_span.span_id) ..
+        "-" .. (proxy_span.should_sample and "1" or "0") ..
+        (proxy_span.parent_id and "-" .. to_hex(proxy_span.parent_id) or ""))
```

Comment:
```suggestion
    set_header("b3", table.concat({
        to_hex(proxy_span.trace_id),
        to_hex(proxy_span.span_id),
        (proxy_span.should_sample and "1" or "0"),
        -- fine if parent_id is nil
        proxy_span.parent_id,
    }, "-"))
```

Reply:
Did a test and seems the `table.concat` has lower performance.
https://gist.github.com/mayocream/8c48b5a27bedb3409434a118bff2120a

Reply:
This way it has better readablity.